scripts/sanitycheck

```diff
@@ -1159,8 +1159,11 @@ class MakeGenerator:
 
             cmd = ["make", "-k", "-j",
                    str(int(CPU_COUNTS * loadmul)), "-f", tf.name, "all"]
+            # assure language neutral environemnt
+            make_env = os.environ.copy()
+            make_env['LC_ALL'] = 'C'
```

Comment:
This will also turn the specified encoding (`LC_CTYPE`) into ASCII for all utilities that are called from `make` that respect it.

Is that okay? Might be best to just set `LC_MESSAGES`.

Reply:
As you can see that produces errors. 

@nashif 
What is the required language setting for sanitycheck? 

LC_TYPE = ?
LC_MESSAGE = ?

Reply:
The problem is that `LC_ALL` overrides `LC_CTYPE` (it overrides all the other `LC_*` variables), which changes the encoding to ASCII.

If you just want to make sure messages from e.g. GCC are printed in English, do `make_env['LC_MESSAGES'] = 'C'`.

Reply:
Unicode and locales is a bit messy by the way, because there's no universal UTF-8 locale that's available everywhere.

If there was, it would make sense to set `LC_ALL` to it if you wanted Unicode support along with the same settings everywhere. Best thing you can do now is to avoid touching any UTF-8 locale settings that might already be there I think, by just settings what you need.

Reply:
Thank you for all the info. This became now try and error as my environment is different from the shippable environment in an unknown way. Using only LC_MESSAGES does not work for me. I get a localized make error message in this case so I stick with LC_ALL. Some tools seem to expect UTF-8 so I switched to C.UTF-8. See what happens.

Reply:
Hmm... maybe `LC_ALL` is already set in your environment, overriding it.

C.UTF-8 is available on most *nixes at least, and Python on Windows already forces UTF-8 by default iirc. Not sure about MacOS though.

Reply:
Passes CI. So LC_ALL=C.UTF-8 is my best bet now.

Reply:
Hmm... if `LC_ALL` was set, changing `LANG` shouldn't have worked though. `LANG` acts as a fallback if some `LC_*` variable isn't set (and `LC_ALL` isn't set).

What does `env | grep "LANG\|LC"` say on your system?

Setting `LC_MESSAGES` works here:

```
LC_MESSAGES=C gcc x
gcc: error: x: No such file or directory
gcc: fatal error: no input files
compilation terminated.

$ LC_MESSAGES=sv_SE.UTF-8 gcc x
gcc: error: x: Filen eller katalogen finns inte
gcc: fatal error: no input files
compilation terminated.
```

Reply:
> env | grep "LANG\|LC"

LANG=de_DE.UTF-8
ZEPHYR_TOOLCHAIN_VARIANT=zephyr
LANGUAGE=

Reply:
The goal should be to have the same test environment regardless of the configuration of the underlying platform. By LC_ALL more configuration space is forced to a defined value. This might be a benefit compared to only change the minimal set of environment variables.

Reply:
@b0661 
Problem is it would give you a non-UTF-8 locale on systems that don't have `C.UTF-8` available, giving that crash you saw earlier.

Are you sure `LC_MESSAGES=C gcc x` vs `LC_MESSAGES=de_DE.UTF-8 gcc x` doesn't work like it does in the example above?

Reply:
@ulfalizer 
> Problem is it would give you a non-UTF-8 locale on systems that don't have C.UTF-8 available, giving that crash you saw earlier.

I see, LC_ALL might make the situation even worse.

What about LC_MESSAGES='C.UTF-8'?

This is what the python code is expecting - UTF-8 in English. If a C.UTF-8 locale is not available it will revert to the default C locale - aka. ASCII, which is valid UTF-8 anyway.